shardus_net/src/header/header_v1.rs

```diff
@@ -72,7 +75,10 @@ impl HeaderV1 {
         let mut sender_id_len_bytes = [0u8; 4];
         cursor.read_exact(&mut sender_id_len_bytes).ok()?;
         let sender_id_len = u32::from_le_bytes(sender_id_len_bytes);
-
+        if sender_id_len > BUFFER_SIZE_LIMIT_IN_BYTES {
+            error!("Sender id size exceeds the limit of {} bytes", BUFFER_SIZE_LIMIT_IN_BYTES);
+            return None;
```

Comment:
Are we also returning None in other places when using similar methods when the condition is not met?

Reply:
For the receive function, we want it to continue listening for other requests as well. Hence over there in the main flow it is continuing.
But for the child functions like `deserialize` it is a one time call made in the receive flow. The intended behaviour for the deserialize function is to return `Some` value or an error (None). I think returning None here makes sense because the size is greater than the limit allowed and hence we should return an error here. 
![image](https://github.com/user-attachments/assets/2e29dabe-a780-4b6c-a556-916736926270)

Reply:
okay, thanks!